ops/_main.py

```diff
@@ -482,7 +466,7 @@ def _make_framework(self, dispatcher: _Dispatcher):
         # If we are in a RelationBroken event, we want to know which relation is
         # broken within the model, not only in the event's `.relation` attribute.
 
-        if self._juju_context.dispatch_path.endswith('-relation-broken'):
+        if self._juju_context.dispatch_path.endswith(('-relation-broken', '_relation_broken')):
```

Comment:
@james-garner-canonical [pushed back on this](https://github.com/canonical/operator/pull/1467/files#r1861280647).

Reply:
I'm not strongly opposed though. I don't know the full reasoning why scenario uses snake here

Reply:
> I don't know the full reasoning why scenario uses snake here

Snake can't be use for things like `ctx.on.relation_broken()` and Scenario tries to have everything consistent, so stuck with snake case throughout, basically normalising everything that way, rather than trying to always be correct in figuring out whether a piece of code was dealing with a name from Juju that's kebab or something from code that's snake.

I like the 'let's normalise and use one case internally' approach, but it does lead to issues like this where that isn't universally applied - JujuContext has exactly what we get from Juju, not a normalised version.

It probably would be cleanest to adjust this in Scenario, but I'm not sure how much work that would be. I'll have a look and see what the best path is - maybe change things in this PR, maybe open an issue for doing it later, maybe something else.

Reply:
Yeah I can see the case for any of the three outcomes here
1. make scenario the same as ops/juju
2. ops normalises everything too
3. we just work around the minor differences, like your original solution here

I think 3 would easily be preferred if 1 or 2 require too much work or would raise issues with backwards compatibility. I certainly don't object to going with 3 for now